qlty-cli/src/commands/coverage/complete.rs

```diff
@@ -0,0 +1,403 @@
+        }
+        if let Some(override_commit_sha) = &self.override_commit_sha {
+            eprintln_unless!(
+                self.quiet,
+                "    override-commit-sha: {}",
+                override_commit_sha
+            );
+        }
+        if let Some(override_pr_number) = &self.override_pr_number {
+            eprintln_unless!(self.quiet, "    override-pr-number: {}", override_pr_number);
+        }
+        if let Some(project) = &self.project {
+            eprintln_unless!(self.quiet, "    project: {}", project);
+        }
+
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_metadata(&self, plan: &Plan) {
+        self.print_section_header(" METADATA ");
+        if !plan.metadata.ci.is_empty() {
+            eprintln_unless!(self.quiet, "    CI: {}", plan.metadata.ci);
+        }
+
+        eprintln_unless!(self.quiet, "    Commit: {}", plan.metadata.commit_sha);
+        if !plan.metadata.pull_request_number.is_empty() {
+            eprintln_unless!(
+                self.quiet,
+                "    Pull Request: #{}",
+                plan.metadata.pull_request_number
+            );
+        }
+
+        if !plan.metadata.branch.is_empty() {
+            eprintln_unless!(self.quiet, "    Branch: {}", plan.metadata.branch);
+        }
+
+        if !plan.metadata.build_id.is_empty() {
+            eprintln_unless!(self.quiet, "    Build ID: {}", plan.metadata.build_id);
+        }
+
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_authentication_info(&self, token: &str) {
+        let token_type = if token.starts_with(COVERAGE_TOKEN_WORKSPACE_PREFIX) {
+            "Workspace Token"
+        } else if token.starts_with(COVERAGE_TOKEN_PROJECT_PREFIX) {
+            "Project Token"
+        } else if let Ok(oidc_regex) = Regex::new(OIDC_REGEX) {
+            if oidc_regex.is_match(token) {
+                "OIDC"
+            } else {
+                "Unknown"
+            }
+        } else {
+            "ERROR"
+        };
+        eprintln_unless!(self.quiet, "    Auth Method: {}", token_type);
+        eprintln_unless!(self.quiet, "    Token: {}", token);
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_complete_success(&self, elapsed_seconds: f32) {
+        eprintln_unless!(
+            self.quiet,
+            "    Coverage marked as complete in {:.2}s!",
+            elapsed_seconds
+        );
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn load_auth_token(&self) -> Result<String> {
+        self.expand_token(match &self.token {
+            Some(token) => Ok(token.to_owned()),
+            None => std::env::var("QLTY_COVERAGE_TOKEN").map_err(|_| {
+                anyhow::Error::msg("QLTY_COVERAGE_TOKEN environment variable is required.")
+            }),
+        }?)
+    }
+
+    fn request_complete(
+        &self,
+        metadata: &qlty_types::tests::v1::CoverageMetadata,
+        token: &str,
+    ) -> Result<Value> {
```

Comment:
The qlty-cloud Crate is intended to be the thing that talks to the API. It defines the QltyClient struct.

Feels like this very particular error handling code should be pushed down into qlty-cloud crate